wallet/src/send_request/mod.rs

```diff
@@ -279,17 +287,91 @@ impl SendRequest {
         take(&mut self.fees)
     }
 
-    pub fn into_partially_signed_tx(self) -> WalletResult<PartiallySignedTransaction> {
+    pub fn into_partially_signed_tx(
+        self,
+        additional_info: &BTreeMap<PoolOrTokenId, UtxoAdditionalInfo>,
+    ) -> WalletResult<PartiallySignedTransaction> {
         let num_inputs = self.inputs.len();
-        let tx = Transaction::new(self.flags, self.inputs, self.outputs)?;
         let destinations = self.destinations.into_iter().map(Some).collect();
+        let utxos = self
+            .utxos
+            .into_iter()
+            .map(|utxo| {
+                utxo.map(|utxo| {
+                    let additional_info = find_additional_info(&utxo, additional_info)?;
+                    Ok(UtxoWithAdditionalInfo::new(utxo, additional_info))
+                })
+                .transpose()
+            })
+            .collect::<WalletResult<Vec<_>>>()?;
+        let output_additional_infos = self
+            .outputs
+            .iter()
+            .map(|utxo| find_additional_info(utxo, additional_info))
```

Comment:
So yeah this should be None or even an error because it's not spendable and cannot be a utxo.